solutions/devsprint-bruno-ramos-2/DeliveryAppChallengeTests/Modules/Product/RestaurantList/RestaurantListInteractorTests.swift

```diff
@@ -11,4 +11,51 @@ import XCTest
 
 class RestaurantListInteractorTests: XCTestCase {
 
+    private let presenterSpy = RestaurantListPresentationLogicSpy()
+    private let workerSpy = RestaurantListWorkingSpy()
+    
+    private lazy var sut = RestaurantListInteractor(presenter: presenterSpy,
+                                                    worker: workerSpy)
+    
+    func test_fetchRestaurantList_whenWorkerReturnsSuccess_shouldCallPresenterWithSuccessError() {
+        // Given
+        workerSpy.fetchRestaurantListCompletionToBeExecuted = .success([.fixture(name: "Padaria Bolo Fofo"), .fixture(name: "Baita Padaria")])
+        
+        // When
+        sut.fetchRestaurantList(request: .init())
+        
+        // Then
+        XCTAssertTrue(workerSpy.fetchRestaurantListCalled)
+        XCTAssertTrue(presenterSpy.presentFetchedRestaurantListCalled)
+        
+        switch presenterSpy.presentFetchedRestaurantListResponsePassed {
+        case .success(let restaurantList):
+            XCTAssertNotNil(restaurantList)
+            XCTAssertEqual(restaurantList[0].name, "Padaria Bolo Fofo")
```

Comment:
Ao invés de utilizarmos o método subscript da API de array do swift, podemos utilizar o método contains de collection extension, dessa forma a gente garante que ao invés de corrermos o risco de dar crash na execução da nossa suíte de testes, ele apenas vai falhar o teste.
```suggestion
            XCTAssertTrue(restaurantList.contains { $0.name "Padaria Bolo Fofo"})
```